### ISSUE TITLE
Translate component

### ISSUE THREAD

**Issue by flowrrs** [i18next/react-i18next#1532]
## Feature

A `<Translate ... />` component.

## Motivation

I recently had a conversation in which I a code screenshot was shared and somebody asked what the `<Trans />` component was.

As a (mostly closeted) trans person, myself, this was an awkward moment for me lol. But I can't suggest that this be improved based on my internalized awkward feelings.

I can, though, use those feelings to justify my initial contemplation of this component and why it was named in such a way that it makes an anime-sized nervous sweat droplet emerge from my forehead. Yes, I admit, my intent originated as a weird bias.

I began pondering this component and digging through source code and came to this question that I just could not answer:

> "Why is it named Trans? (😅)"

For the past year I have lead a team that struggles with maintaining our codebase due to the effects on readability that 2-3 years of poorly naming things (among other code quality concerns) has had.

So I am a very big proponent of my philosophy and core approach to improving code quality: **"name it what it is"**.

I can not determine why this component would be exported as `Trans` and not `Translate`. (Just as I can not determine why `t` is not `translate`, but I am not ready to fight that fight yet. 🙃)

"Translate" is not a big word that would warrant abbreviation.

"Translate" is not a common abbreviation for "translate".

Naming this component `T` would have made more sense than `Trans`, because at least it would be more consistent with `t` from i18n.

I'd like to propose that it be exported with an alias of `Translate` in order to be more _trans_parent, intuitive, and promote better code quality globally.

When developing tools to be consumed by others, we should be concerned with developer experience. If you see the image in this collapsible thing below, you can see that Cody didn't understand what `Trans` meant.

A good DX, like UX, should enable at anybody, globally, to be able to use a tool without finding themselves scratching their head.

<details>
<summary>
[Expand me] Group chat with a guy I taught to code and got into the industry in 2017, and another guy that I am currently teaching.
</summary>
<img src="https://user-images.githubusercontent.com/104879649/177630760-553a0c8d-d8f4-4641-9a1b-e19648156f20.png"/>
</details>

## Example

```jsx
import React from 'react';
import { Translate, useTranslation } from 'react-i18next'

function MyComponent() {
  const { t } = useTranslation('myNamespace');

  return <Translate t={t}>Hello World</Tranlslate>;
}
```

**Comment by adrai** [comment]
tldr; it is named [Trans](https://react.i18next.com/latest/trans-component) because [Translation](https://react.i18next.com/latest/translation-render-prop) already existed. And the alternative of `T` was confusing when speaking of passing `t` to `T`.

**Comment by jamuhl** [comment]
That `Trans` component was introduced at some point other modules already decided about naming it that way:

- https://github.com/lingui/js-lingui#example

The `t` function naming exists for so long...and is used all over the place.

Beside `t` function there are also `trans` functions

- Laravel trans: https://laravel.com/docs/5.0/localization#basic-usage
- vue trans: https://www.npmjs.com/package/vue-trans

It is and for sure not was our intent to make people feel awkward.

**Comment by flowrrs** [comment]
It was just humor to build the narrative around the issue, that is all. :)

But I understand your reasoning. I appreciate you!
